### ISSUE TITLE
run: v0.3 phase 3 teach model to say "i cannot hear/understand you"

### ISSUE THREAD

**Issue by bachvudinh** [homebrewltd/ichigo#66]
## Goal
See: #56 

## Methodology
- In this phase, we fine-tune the model from the phase 2 checkpoint using various LoRA rank and alpha settings. Our goal is to introduce new tasks to the model while preserving its existing knowledge. We experiment with different LoRA configurations to optimize the balance between task adaptation and knowledge retention.
- Data Source: Using the background noise dataset from : https://huggingface.co/datasets/Myrtle/CAIMAN-ASR-BackgroundNoise and duplicate each audio 10 times, then i use whisperVQ to tokenize these audio leading to around 11k samples. Lastly, i created diverse 10 LLM's responses with overall content is that there is no sementic word in this audio so it refuse to answer the querry.

## Hyperparams
Parameter | Value
--- | ---
Rank | Vary from 2-64
Alpha | 
Epoch | 1
Global batch size | 32
Learning Rate | 8e-6
Learning Scheduler | Cosine
Optimizer | AdamW Fused
Warmup Steps | 7
Weight Decay | 0.01
Gradient Checkpointing | Full
Max length | 4096
Precision | bf16

## Results
- MMLU:
  + Checkpoint r=64, alpha=128, target modules not containing lm_head and embed_tokens layer:
  
     ![Image](https://github.com/user-attachments/assets/c491a313-dc3f-42a2-9121-3485500abcc2)

  + Checkpoint r=32, alpha=64:

    ![Image](https://github.com/user-attachments/assets/d3ef8984-c88a-430e-a15d-931f0a0e6dfc)

  + Checkpoint r=8, alpha=16:

    ![Image](https://github.com/user-attachments/assets/f1e8ca98-8c27-47e2-8207-8a9c1faa3692)

  + Checkpoint r=8, alpha=16, target modules not containing lm_head and embed_tokens layer:

    ![Image](https://github.com/user-attachments/assets/a5fe2f59-1982-4f7b-b8e9-4e31da568f74)

  + Checkpoint r=4, alpha=4:

    ![Image](https://github.com/user-attachments/assets/05f12c75-c741-4e70-8d46-a7dd9ee2fec9)

  + Checkpoint r=4, alpha=4 but trained with only half of data:
  
      ![Image](https://github.com/user-attachments/assets/5ce13476-0dd6-419c-b80e-5f1d5b57c5cc)

   + Checkpoint r=4, alpha=8, target modules not containing lm_head and embed_tokens layer:
  
      ![Image](https://github.com/user-attachments/assets/00b0d201-5c97-4891-b05e-0aafe3c241c0)
  
- Audio-Bench:
  + Speech Instruction (GPT-4 judge):
    ** Checkpoint r=64, alpha=128, target modules not containing lm_head and embed_tokens layer:
    Alpaca : 2.07
    Open-hermes: 2.32
    
  
  + ASR (WER score):
    ** Checkpoint r=64, alpha=128, target modules not containing lm_head and embed_tokens layer:
    Score~ 1.00
    

## Learnings
- Model return repetition response when processing the low quality audio

## Quicklinks
- Data: 
  + Background noise data: https://huggingface.co/datasets/homebrewltd/instruction-background-noise-data. 

- Checkpoints: 
  + All the experiment checkpoint you can find at : https://huggingface.co/jan-hq at Model Section.

**Comment by bachvudinh** [comment]
The updated results of 23-09-2024 experiments:

**Comment by bachvudinh** [comment]
#71 The details for final training data here.

**Comment by tikikun** [comment]
We experienced some catastrophic failure with 3 runs at the same time, delay this to next milestone.

**Comment by bachvudinh** [comment]
# The updated results of 23-09-2024 experiments:

## Data Distribution:
- The details of data is here #71  but we regenerate the speech multiturn data with the right checkpoint.

## Training config:
- Cause we mix it with 80% phase 2 training data so we move to full-finetune.
- It took ~ 1 hours on 8xH100.

Parameter | Value
--- | ---
Epoch | 1
Global batch size | 256
Learning Rate | 1.5e-5
Learning Scheduler | Cosine
Optimizer | AdamW Fused
Warmup Steps | 8
Weight Decay | 0.01
Gradient Checkpointing | Full
Max length | 4096
Precision | bf16
## Results:
- MMLU score:
 
![Screenshot from 2024-09-28 21-18-46](https://github.com/user-attachments/assets/4a908dfd-04dd-4834-b4f7-7338e358fa96)

- Audio-Bench:
  + Speech Instruction (GPT-4 judge):
    Alpaca : 3.63
    Open-hermes: 3.64
    
  + ASR (WER score):
    Score~ 0.28

## Current Model Problem
- We successfully teach the model to refuse answering noise audio. Furthermore, we improve the multiturn ability of the model and maintaining model's performance on MMLU and AudioBench.
- But Model still sometime respose to noise audio in long conversations. This is expected because we dont have the multiturn noise data yet--> will improve in the next version
